File: baseLibrary/src/main/java/cn/com/base/simple/bean/ImageModel.java
package cn.com.base.simple.bean;

import android.databinding.BindingAdapter;
import android.widget.ImageView;

import cn.com.base.glide.GlideUtil;
import cn.com.base.simple.util.StrUtils;

/**
 * author      : wantao
 * date        : 2017/11/8 14:25
 * description : 图片加载类
 */
abstract public class ImageModel{

    protected String imageUrl;

    @BindingAdapter({"imageUrl"})
    public static void loadImage(ImageView view, String imageUrl) {
        GlideUtil.getInstance().loadCommonImg(view.getContext(), StrUtils.getResourceUrl(imageUrl),view);
    }

    abstract public String getImageUrl();

    @BindingAdapter({"imgurl"})
    public static void loadurlImg(ImageView view, String imageUrl){
        GlideUtil.getInstance().loadCommonImg(view.getContext(),imageUrl,view);
    }
}

File: baseLibrary/src/main/java/cn/com/base/simple/databinding/ImageModel.kt
package cn.com.base.simple.databinding

import android.databinding.BindingAdapter
import android.widget.ImageView
import cn.com.base.glide.GlideUtil
import cn.com.base.simple.util.StrUtils

/**
 * author      : wantao
 * date        : 2017/11/8 14:25
 * description : 图片
 */
open  class ImageModel {

    /**
     * 图片加载
     *
     * @param view
     * @param imageUrl
     */
    @BindingAdapter("imageUrl")
    fun loadImage(view: ImageView, imageUrl: String) {
        GlideUtil.getInstance().loadCommonImg(view.context, StrUtils.getResourceUrl(imageUrl), view)
    }

}